jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceWithEntitiesTest.java

```diff
@@ -0,0 +1,239 @@
+        assertTrue(isDeployed);
+
+        assertNotNull(processService);
+
+        Map<String, Object> params = new HashMap<String, Object>();
+
+        long processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "processvarentity", params);
+        assertNotNull(processInstanceId);
+
+        ProcessInstance pi = processService.getProcessInstance(processInstanceId);
+        assertNotNull(pi);
+
+        Map<String, Object> variables = processService.getProcessInstanceVariables(processInstanceId);
+        assertNotNull(variables);
+        assertEquals(1, variables.size());
+        assertTrue(variables.containsKey("caseDetails"));
+
+        VariableEntity varEntity = (VariableEntity) variables.get("caseDetails");
+
+        // make sure getting mapped variables does NOT throw LazyInitializationException
+        try {
+            varEntity.getMappedVariables();
+        } catch(LazyInitializationException e ) {
+            fail("Unable to retrieve mapped variables : " + e.getMessage());
+        }
+
+    }
+
+    @After
+    public void cleanup() {
+        cleanupSingletonSessionId();
+        if (units != null && !units.isEmpty()) {
+            for (DeploymentUnit unit : units) {
+                try {
+                    deploymentService.undeploy(unit);
+                } catch (Exception e) {
+                    // do nothing in case of some failed tests to avoid next test to fail as well
+                }
+            }
+            units.clear();
+        }
+        close();
+    }
+
+    private static String getPersistenceXml() {
+        return "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n" +
+                "<persistence xmlns=\"http://java.sun.com/xml/ns/persistence\" xmlns:orm=\"http://java.sun.com/xml/ns/persistence/orm\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" version=\"2.0\" xsi:schemaLocation=\"http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd http://java.sun.com/xml/ns/persistence/orm http://java.sun.com/xml/ns/persistence/orm_2_0.xsd\">\n" +
+                "    <persistence-unit name=\"org.jbpm.test:test-module:1.0.0-SNAPSHOT\" transaction-type=\"JTA\">\n" +
+                "        <provider>org.hibernate.ejb.HibernatePersistence</provider>\n" +
+                "        <jta-data-source>java:jdbc/testDS1</jta-data-source>\n" +
+                "        <class>example.CaseDetail</class>\n" +
+                "        <class>org.drools.persistence.jpa.marshaller.MappedVariable</class>\n" +
+                "        <class>org.drools.persistence.jpa.marshaller.VariableEntity</class>\n" +
+                "        <exclude-unlisted-classes>true</exclude-unlisted-classes>\n" +
+                "        <properties>\n" +
+                "            <property name=\"hibernate.dialect\" value=\"org.hibernate.dialect.PostgreSQLDialect\"/>\n" +
```

Comment:
@tsurdilo I have one question. Why is this persistence.xml in String and not in a seperate resource file, moreover, is it mentioned to be run only with PostgreSQL?